tests/unit/s2n_client_hello_retry_test.c

```diff
@@ -176,6 +176,7 @@ int main(int argc, char **argv)
                 .kem_preferences = &kem_preferences_all,
                 .signature_preferences = &s2n_signature_preferences_20200207,
                 .ecc_preferences = &s2n_ecc_preferences_20200310,
+                .strongly_preferred_groups = &s2n_supported_group_preferences_null,
```

Comment:
Do you need to add this field to every single test policy? I don't remember having to do that when I added a new field on the security policy. Doesn't a null pointer signify no strongly supported groups?

This would additionally massively cut down the number of changes in this PR.

Reply:
> Do you need to add this field to every single test policy?

I don't *need to*, but I think it's safer to do it this way, as it acts as a safety net if there's ever a forgotten null check on `strongly_preferred_groups`. 

It's slightly more work for me to add `&s2n_supported_group_preferences_null` everywhere, but it avoids the possibility of `policy->strongly_preferred_groups->count` ever causing a SegFault due to `strongly_preferred_groups` being null.